test/Table_Tests/src/In_Memory/Table_Conversion_Spec.enso

```diff
@@ -0,0 +1,158 @@
+
+    Test.group "from_objects with Array" <|
+        Test.specify "Generates a table from a mixed vector of single values and JSON objects, with missing and extra fields" <|
+            expected = Table.from_rows ["first", "last", "blah", "Value"] [["Mary", "Smith", Nothing, Nothing], ["Joe", "Burton", Nothing, Nothing], [Nothing, Nothing, Nothing, 12]]
+            Table.from_objects (uniform_json+[12]).to_array ["first", "last", "blah", "Value"] . should_equal expected
+
+    Test.group "expand_column" <|
+        Test.specify "Expands a column of single values" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", [3, 4]], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb Value", [3, 4]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" . should_equal expected
+
+        Test.specify "Expands a uniform column of JSON objects" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", uniform_json], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb first", ["Mary", "Joe"]], ["bbb last", ["Smith", "Burton"]], ["bbb age", [23, 34]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" . should_equal expected
+
+        Test.specify "Expands a uniform column of JSON objects, with extra and missing fields" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", uniform_json], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb age", [23, 34]], ["bbb foo", [Nothing, Nothing]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" ["age", "foo"] . should_equal expected
+
+        Test.specify "Expands a non-uniform column of JSON objects" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", non_uniform_json], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb first", ["Mary", Nothing]], ["bbb last", ["Smith", Nothing]], ["bbb age", [23, Nothing]], ["bbb height", [Nothing, 1.9]], ["bbb weight", [Nothing, 70]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" . should_equal expected
+
+        Test.specify "Expands a non-uniform column of JSON objects with extra and missing fields" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", non_uniform_json], ["ccc", [5, 6]]]
+            expected = Table.new [["aaa", [1, 2]], ["bbb last", ["Smith", Nothing]], ["bbb height", [Nothing, 1.9]], ["bbb foo", [Nothing, Nothing]], ["ccc", [5, 6]]]
+            table.expand_column "bbb" ["last", "height", "foo"] . should_equal expected
+
+        Test.specify "Error if there are vectors/arrays within a column" <|
+            table = Table.new [["aaa", [1, 2]], ["bbb", [[1, 2, 3], [4, 5, 6]]]]
+            table2 = Table.new [["aaa", [1, 2]], ["bbb", [[1, 2, 3].to_array, [4, 5, 6].to_array]]]
+            table.expand_column "bbb" . should_fail_with Illegal_Argument
+            table2.expand_column "bbb" . should_fail_with Illegal_Argument
```

Comment:
For now this is good as this documents the existing behaviour.

But I'm wondering if that's what we want to keep - I'm not sure if Vector really differs from any other scalar Value to error instead to just be 'not expanded' this way. But that is for later.